war/src/main/js/pluginSetupWizardGui.js

```diff
@@ -0,0 +1,555 @@
+		var top = $pl.scrollTop() + $el.position().top;
+		$pl.stop(true).animate({
+			scrollTop: top
+		}, 250, function() {
+			var top = $pl.scrollTop() + $el.position().top;
+			$pl.stop(true).scrollTop(top);
+		})
+	};
+	
+	// handle show/hide details during the installation progress panel
+	var toggleInstallDetails = function() {
+		var $c = $('.install-console');
+		if($c.is(':visible')) {
+			$c.slideUp();
+		}
+		else {
+			$c.slideDown();
+		}
+	};
+	
+	// scoped click handler, prevents default actions automatically
+	var bindClickHandler = function(cls, action) {
+		$wizard.on('click', cls, function(e) {
+			action.apply(this, arguments);
+			e.preventDefault();
+		});
+	};
+	
+	// click action mappings
+	var actions = {
+		'.install-recommended': installDefaultPlugins,
+		'.install-custom': loadCustomPluginPanel,
+		'.install-home': function() { setPanel(welcomePanel); },
+		'.install-selected': function() { installPlugins(selectedPlugins); },
+		'.toggle-install-details': toggleInstallDetails,
+		'.install-done': finishInstallation,
+		'.plugin-select-all': function() { selectedPlugins = recommendedPlugins.slice(0); refreshPluginSelectionPanel(); },
+		'.plugin-select-none': function() { selectedPlugins = []; refreshPluginSelectionPanel(); },
+		'.plugin-select-recommended': function() { selectedPlugins = installData.defaultPlugins.slice(0); refreshPluginSelectionPanel(); },
+		'.plugin-show-selected': toggleSelectedSearch,
+		'.select-category': selectCategory
+	};
+	for(var cls in actions) {
+		bindClickHandler(cls, actions[cls]);
+	}
+	
+	// kick off to get resource bundle
+	jenkins.get('/i18n/resourceBundle?baseName=jenkins.install.pluginSetupWizard', function(res) {
+		if(res.status != 'ok') {
+			msg = { error: 'Unable to load localization data: ' + res.message };
+			setPanel(errorPanel);
+			return;
+		}
+		
+		msg = res.data;
+		
+		// check for updates when first loaded...
+		jenkins.get('/updateCenter/api/json?tree=jobs[name,type,status[*]]', function(data) {
+			// check for install jobs
+			for(var i = 0; i < data.jobs.length; i++) {
+				if(data.jobs[i].type != 'ConnectionCheckJob') {
+					showInstallProgress();
+					return;
+				}
+			}
+			
+			// If no active install, by default, we'll show the welcome screen
+			setPanel(welcomePanel);
+		});
+		
+		// check the connectivity api
+		var testConnectivity = function() {
+			jenkins.get('/updateCenter/connectionStatus?siteId=default', function(response) {
+				var uncheckedStatuses = ['CHECKING', 'UNCHECKED'];
+				if(uncheckedStatuses.indexOf(response.data.updatesite) >= 0  || uncheckedStatuses.indexOf(response.data.internet) >= 0) {
+					setTimeout(testConnectivity, 500);
+				}
+				else {
+					if(response.status != 'ok' || response.data.updatesite != 'OK' || response.data.internet != 'OK') {
+						setPanel(offlinePanel);
+					}
+				}
+			});
+		};
+		testConnectivity();
+	});
+};
+
+// export wizard creation method
+exports.init = createPluginSetupWizard;
```

Comment:
Maybe this is ok for now but I'd really like to see it modularised a bit. As it stands, we basically have the whole thing in a single .JS file.